hw/ip/otbn/util/shared/operand.py

```diff
@@ -309,47 +312,56 @@ def op_val_to_enc_val(self,
         if self.width is None:
             return None
 
+        # The operand can have two offsets applied to it. The PC offset is
+        # applied before shifting, the fixed_offset is applied after shifting.
         if self.pc_rel:
             # If the operand is PC-relative and we don't have cur_pc, give up.
             if cur_pc is None:
                 return None
 
             # Otherwise, we need to encode the offset.
-            offset_val = op_val - cur_pc
+            pc_offset_val = op_val - cur_pc
         else:
-            offset_val = op_val
+            pc_offset_val = op_val
 
-        # First, try to shift right. Check that we won't clobber any low bits.
+        # Following PC offset, try to shift right. Check that we won't clobber
+        # any low bits.
         shift_mask = (1 << self.shift) - 1
-        if offset_val & shift_mask:
+        if pc_offset_val & shift_mask:
             raise ValueError('Cannot encode the value {}: the operand has a '
                              'shift of {}, but that would clobber some bits '
                              '(because {} & {} = {}, not zero).'
-                             .format(offset_val, self.shift,
-                                     offset_val, shift_mask,
-                                     offset_val & shift_mask))
+                             .format(pc_offset_val, self.shift,
+                                     pc_offset_val, shift_mask,
+                                     pc_offset_val & shift_mask))
 
-        shifted = offset_val >> self.shift
+        shifted = pc_offset_val >> self.shift
+
+        # Finally apply the fixed offset
+        final_val = shifted - self.fixed_offset
```

Comment:
With the renaming I suggested above, this could now be called `offset_val`, which is maybe a bit more descriptive.